mailserver/mailserver.go

```diff
@@ -124,41 +146,47 @@ func (s *WMailServer) Close() {
 	}
 }
 
-// Archive a whisper envelope
+// Archive a whisper envelope.
 func (s *WMailServer) Archive(env *whisper.Envelope) {
 	key := NewDbKey(env.Expiry-env.TTL, env.Hash())
 	rawEnvelope, err := rlp.EncodeToBytes(env)
 	if err != nil {
 		log.Error(fmt.Sprintf("rlp.EncodeToBytes failed: %s", err))
 	} else {
-		err = s.db.Put(key.raw, rawEnvelope, nil)
-		if err != nil {
+		if err = s.db.Put(key.raw, rawEnvelope, nil); err != nil {
 			log.Error(fmt.Sprintf("Writing to DB failed: %s", err))
 		}
 	}
 }
 
-// DeliverMail sends mail to specified whisper peer
+// DeliverMail sends mail to specified whisper peer.
 func (s *WMailServer) DeliverMail(peer *whisper.Peer, request *whisper.Envelope) {
 	if peer == nil {
 		log.Error("Whisper peer is nil")
 		return
 	}
+	s.managePeerLimits(peer.ID())
```

Comment:
It looks like we do not stop processing the request if the peer is exceeding its limit.